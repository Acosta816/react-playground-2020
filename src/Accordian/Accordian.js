import React from 'react';


class Accordian extends React.Component {

    static defaultProps = {
        sections: []
    }

    state = {
        currentTab: 0
    }

    renderButtons = ()=> {
        const myButtons = this.props.sections.map((section,index)=> {
            return(
                <li key={index}>
                    <button onClick={()=>{this.changeState(index)}} >{section.name}</button>
                    { index === this.state.currentTab ? this.renderContent(index) : null }
                </li>
            )
        })

        return myButtons;
    }

    renderContent = (index)=> {
        return (
            <div>
                <p>{this.props.sections[index].content}</p>
                <img src={this.props.sections[index].avatar} alt="sonic gifs" />
            </div>
        )
    }

    changeState = (index)=> {
        console.log(index);
        this.setState({
            currentTab: index
        })
    }

    render() {

        return(
            <div>
                <ul>
                    {this.props.sections.length ? this.renderButtons() : null }
                </ul>
            </div>
        )
    }
}


export default Accordian;