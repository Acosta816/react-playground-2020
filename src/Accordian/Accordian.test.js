import React from 'react';
import ReactDOM from 'react-dom';
import Accordian from './Accordian';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('TestSuite: Accordian Component', ()=> {

    //declare all test variables here
    const sections = [{name:'Sonic', content: 'Fastest thing Alive', avatar: "https://media2.giphy.com/media/K3B3v1IUkroE8/giphy.gif?cid=790b7611a0141074eddd3986bac2408e78aa2c9921709e38&rid=giphy.gif"}, {name:'Tails', content: 'Loyal and Hyper-intelligent Flying Fox', avatar: "https://66.media.tumblr.com/a2b506d774c4840d65527670dd77076e/tumblr_pc8gylOts11qjztgpo1_400.gifv"}, {name:'Amy', content: 'Sonics Biggest Fan. Hammer-weilding Pink Hedgehog', avatar: 'https://appstickers-cdn.appadvice.com/454316134/827794092/56d7cab123afdb67c732b5ceb21206ef-1.gif'}, {name:'Robotnik', content: 'Crazed Genius, obsessed with robotisizing all life and ridding the world of Sonic', avatar: 'https://thumbs.gfycat.com/RepentantPepperyBangeltiger.webp'}, {name:'Shadow', content: 'Sonics adversary. Created by Robotnik to become the ultimate lifeform'}, {name:'Sally', content: 'Rightful Princess of Mobius and one of the Leaders in the Freedom Fighters against Robotnik. Sonics Best friend.'}];

    it('Smoke: renders without crashing', ()=> {
        const div = document.createElement('div');

        ReactDOM.render(<Accordian />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('Snapshot: should render empty, given no sections prop', ()=> {
        const enzWrapper = shallow(<Accordian />);
        expect(toJson(enzWrapper)).toMatchSnapshot();
    })

    it('Snapshot: should render with first tab open', ()=> {
        const enzWrapper = shallow(<Accordian sections={sections} />);
        expect(toJson(enzWrapper)).toMatchSnapshot();
    });

    it('Snapshot: should close first tab and open 4th tab when 4th button is clicked on', ()=> {
        const enzWrapper = shallow(<Accordian sections={sections}/>);

        enzWrapper.find('button').at(3).simulate('click');

        expect(toJson(enzWrapper)).toMatchSnapshot();
    })

    it('Snapshot: should close first tab and open 2nd tab when 2nd button is clicked on', ()=> {
        const enzWrapper = shallow(<Accordian sections={sections}/>);

        enzWrapper.find('button').at(1).simulate('click');

        expect(toJson(enzWrapper)).toMatchSnapshot();
    })



})