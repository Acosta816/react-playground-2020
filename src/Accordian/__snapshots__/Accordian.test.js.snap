// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestSuite: Accordian Component Snapshot: should close first tab and open 2nd tab when 2nd button is clicked on 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Sonic
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Tails
      </button>
      <div>
        <p>
          Loyal and Hyper-intelligent Flying Fox
        </p>
        <img
          alt="sonic gifs"
          src="https://66.media.tumblr.com/a2b506d774c4840d65527670dd77076e/tumblr_pc8gylOts11qjztgpo1_400.gifv"
        />
      </div>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Amy
      </button>
    </li>
    <li
      key="3"
    >
      <button
        onClick={[Function]}
      >
        Robotnik
      </button>
    </li>
    <li
      key="4"
    >
      <button
        onClick={[Function]}
      >
        Shadow
      </button>
    </li>
    <li
      key="5"
    >
      <button
        onClick={[Function]}
      >
        Sally
      </button>
    </li>
  </ul>
</div>
`;

exports[`TestSuite: Accordian Component Snapshot: should close first tab and open 4th tab when 4th button is clicked on 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Sonic
      </button>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Tails
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Amy
      </button>
    </li>
    <li
      key="3"
    >
      <button
        onClick={[Function]}
      >
        Robotnik
      </button>
      <div>
        <p>
          Crazed Genius, obsessed with robotisizing all life and ridding the world of Sonic
        </p>
        <img
          alt="sonic gifs"
          src="https://thumbs.gfycat.com/RepentantPepperyBangeltiger.webp"
        />
      </div>
    </li>
    <li
      key="4"
    >
      <button
        onClick={[Function]}
      >
        Shadow
      </button>
    </li>
    <li
      key="5"
    >
      <button
        onClick={[Function]}
      >
        Sally
      </button>
    </li>
  </ul>
</div>
`;

exports[`TestSuite: Accordian Component Snapshot: should render empty, given no sections prop 1`] = `
<div>
  <ul />
</div>
`;

exports[`TestSuite: Accordian Component Snapshot: should render with first tab open 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
      >
        Sonic
      </button>
      <div>
        <p>
          Fastest thing Alive
        </p>
        <img
          alt="sonic gifs"
          src="https://media2.giphy.com/media/K3B3v1IUkroE8/giphy.gif?cid=790b7611a0141074eddd3986bac2408e78aa2c9921709e38&rid=giphy.gif"
        />
      </div>
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
      >
        Tails
      </button>
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
      >
        Amy
      </button>
    </li>
    <li
      key="3"
    >
      <button
        onClick={[Function]}
      >
        Robotnik
      </button>
    </li>
    <li
      key="4"
    >
      <button
        onClick={[Function]}
      >
        Shadow
      </button>
    </li>
    <li
      key="5"
    >
      <button
        onClick={[Function]}
      >
        Sally
      </button>
    </li>
  </ul>
</div>
`;
